spark/v3.0/spark/src/main/java/org/apache/iceberg/spark/SparkDataFile.java

```diff
@@ -81,6 +82,7 @@ public SparkDataFile(Types.StructType type, StructType sparkType) {
     keyMetadataPosition = positions.get("key_metadata");
     splitOffsetsPosition = positions.get("split_offsets");
     sortOrderIdPosition = positions.get("sort_order_id");
+    schemaIdPosition = positions.get("schema_id");
```

Comment:
Nit: should we put it in the consistent position, after file format?

Reply:
Updated